packages/contracts/src/healthcheck/HealthChecker.sol

```diff
@@ -86,17 +100,76 @@ abstract contract HealthChecker is SafeInitializable, OwnableUpgradeable {
             return;
         }
 
-        // Perform the health check, revert the transaction if unsuccessful.
-        bool success = healthCheck.check(
-            strategy,
-            profit,
-            loss,
-            debtPayment,
-            debtOutstanding,
-            totalDebt
-        );
-        if (!success) {
+        // If no custom health check implementation provided, call default one.
+        uint8 checkResult = 0;
+        if (address(healthCheck) == address(0)) {
+            checkResult = _defaultCheck(
+                strategy,
+                profit,
+                loss,
+                totalDebt
+            );
+        } else {
+            // Perform the health check, to revert the transaction
+            checkResult = healthCheck.check(
+                strategy,
+                profit,
+                loss,
+                debtPayment,
+                debtOutstanding,
+                totalDebt,
+                gasCost
+            );
+        }
+
+        if(checkResult == 2) { // call fallback if loss to big
+            healthCheckFallback();
+        } else if(checkResult == 1) { // call secondary fallback if loss is acceptable
+            acceptableLossFallback();
+        }
+        // just allow to run normally in case of transaction is profitable
+    }
+
+    /// @dev Fallback function that is called when the health check fails.
+    function healthCheckFallback() internal virtual {
+        revert HealthCheckFailed();
+    }
+
+    /// @dev Fallback function that is called when the health check fails but loss is acceptable.
+    /* solhint-disable no-empty-blocks */
+    function acceptableLossFallback() internal virtual {
+        // do nothing by default but can be overridden
+    }
+    /* solhint-disable no-empty-blocks */
+
+    /// @notice Performs the health check by comparing execution strategy parameters. This behavior can be overridden with custom health check implementation.
+    /// @param profit The amount of funds that the strategy realised as profit.
+    /// @param loss The amount of funds that the strategy realised as loss.
+    /// @param totalDebt The total amount of funds borrowed by the strategy from the vault.
+    function _defaultCheck(
+        address strategy,
+        uint256 profit,
+        uint256 loss,
+        uint256 totalDebt
+    ) internal returns (uint8 result)
+    {
+        // If no target provided skipp the execution.
+        if (address(strategy) == address(0)) {
             revert HealthCheckFailed();
         }
+
+        if(profit > 0 || loss == 0) {
+            // if stategy profit is positive or zero, then it is healthy and we need to get this profit
+            result = 0;
+        } else if(loss > 0 && loss <= totalDebt * shutdownLossRatio / MAX_BPS) { //todo verify the condition
+            // if loss is positive but below critical loss threshold
+            result = 1;
```

Comment:
Do you think we need this "intermediate" state? "Loss" - is acceptable event of the strategy behaviour. I personally ok with this, but mind the contract size, our contracts quickly become bloated with unnecessary code.

Reply:
Contract size is a complex topic. Do we want to implement other technics to mitigate this issue? You mentioned diamonds or other solutions another day. I think this small portion of if else won't solve the situation because it is too small.

Reply:
I think we correctly moved this to another contract. One additional if for the intermediate state, is not a big deal, but will save as much when we will want to enhance this logic. We can remove logic which handle acceptable loss on strategy side, it will decrease sise on few lines but it not a big deal. My personal preference keep it as it is for now